app/models/course/registration.rb

```diff
@@ -0,0 +1,29 @@
+class Course::Registration < ActiveRecord::Base
+  stampable
+
+  scope :staff, -> { where(role: roles.values_at(:teaching_assistant, :manager, :owner)) }
```

Comment:
Find a gem which might be useful: https://github.com/brainspec/enumerize

it supports scopes like:

```
   User.with_role(:teaching_assistant, :manger, :owner)
```

And also helps with I18n. It does not support the categories things, but I think it's fine? just define a new scope with the code above.